lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/VortexRequestor.java

```diff
@@ -33,17 +35,19 @@
 @DriverSide
 class VortexRequestor {
   private final ExecutorService executorService = Executors.newCachedThreadPool();
+  private final Codec inputCodec;
 
   @Inject
-  VortexRequestor() {
+  VortexRequestor(@Parameter(VortexMasterConf.InputCodec.class) final Codec inputCodec) {
+    this.inputCodec = inputCodec;
   }
 
   void send(final RunningTask reefTask, final VortexRequest vortexRequest) {
     executorService.execute(new Runnable() {
       @Override
       public void run() {
+        reefTask.send(VortexAvroUtils.toBytes(inputCodec, vortexRequest));
```

Comment:
What happens if serialization/deserialization fails? Will we be able to recover the Tasklets that have failed? Do we want to fail an Evaluator in this case?